R/language.R

```diff
@@ -21,36 +22,115 @@
 #' is_lang(q3)
 #' is_call(q3)
 #'
-#' # A literal is either a scalar atomic vector or a name:
-#' is_literal("string")
-#' is_literal(letters)
-#' is_literal(quote(name))
+#'
+#' # Atomic language objects are the terminating nodes of a call
+#' # tree. They are either a scalar atomic vector, a name, or NULL:
+#' is_atomic_lang("string")
```

Comment:
by the way, I think I understand better the concept of literal now. IIRC, literal refers to anything that's self-evaluable. So `quote(c(1, 2))` is not a literal but the SEXP that results from evaluating this in the base environment is a literal. Another way to view this is that anything that you can evaluate in the empty environment is a literal.

The tricky part is that literal vectors are represented in R in the same way as their language representation.

Since literal vectors are definitely atomic language objects as well (and technically scalar and vectors have the same representation as a wrapped C array), I'm not sure anymore that this predicate represents the right concept. Should we unexport it for now?